File: js/helloworld/helloworld.js
#!/usr/bin/env gjs


// requires the gjs library:
//   https://gitlab.gnome.org/GNOME/gjs

// run with:
//   gjs helloworld.scm


const Gtk = imports.gi.Gtk;


let app = new Gtk.Application({ application_id: "org.gtk.HelloWorld" });

app.connect(
    "activate", () => {
        let win = new Gtk.ApplicationWindow({ application: app });
        let btn = new Gtk.Button({ label: "Hello, World!" });

        btn.connect("clicked", () => { win.close(); });
        win.add(btn);
        win.show_all();
    }
);


app.run([]);

File: erlang/helloworld/helloworld.erl
% -*- erlang -*-


% run with:
%   escript helloworld.erl

% FIXME: Doesn't work
%          - clicking the button doesn't close the app
%          - no main loop


-module(helloworld).

-export([start/0]).
-export([handle_click/1]).

-include_lib("wx/include/wx.hrl").


handle_click(#wx{}) ->
    halt.


start() ->
    wx:new(),

    Frame = wxFrame:new(wx:null(), ?wxID_ANY, "Hello World"),
    Button = wxButton:new(Frame, ?wxID_ANY, [{label, "Hello World!"}]),
    wxButton:connect(
      Button,
      command_button_clicked,
      [{callback, handle_click}]
     ),

    wxFrame:show(Frame),
    wxButton:show(Button),

    timer:sleep(10000).


%% for running with escript
main(_) ->
    start().
